src/CopyrighFixer/CopyrighFixer/sign.cpp

```diff
@@ -37,4 +45,115 @@ const QString &Signature::getMessage() const {
     return _customMessage;
 }
 
+bool Signature::fromJson(const QString &pathToFile) {
+
+    QFileInfo signFile(pathToFile);
+    if (signFile.exists() && signFile.isFile()) {
+
+        QFile file(pathToFile);
+        if (!file.open(QIODevice::ReadOnly)) {
+            QuasarAppUtils::Params::log("Json file couldn't be opened/found");
+            return false;
+        }
+
+        QByteArray byteSignFile(file.readAll());
+        file.close();
+
+        QJsonDocument jsDoc(QJsonDocument::fromJson(byteSignFile));
+        if (jsDoc.isNull()) {
+            QuasarAppUtils::Params::log("Failed to create JSON doc.");
+            return false;
+        }
+        if (!jsDoc.isObject()) {
+            QuasarAppUtils::Params::log("JSON is not an object.");
+            return false;
+        }
+
+        QJsonObject jsObj = jsDoc.object();
+        if (jsObj.isEmpty()) {
+            QuasarAppUtils::Params::log("JSON object is empty.");
+            return false;
+        }
+
+        Owner ownObj;
+        QJsonArray ownLst(jsObj.value("ownersList").toArray());
+        for (auto itemLst = ownLst.cbegin(); itemLst != ownLst.end(); ++itemLst) {
+            ownObj.fromjson(itemLst->toObject());
+            _ownersMap.insert(itemLst->toObject().value("timePoint").toInt(), ownObj);
+        }
+
+        _licenseTitle = jsObj.value("license").toString();
+        _customMessage = jsObj.value("licenseText").toString();
+
+        return true;
+
+    } else {
+        return false;
+    }
+
+}
+
+bool Signature::toJson(QString &pathToFile) const {
+
+    QFileInfo checkFile(pathToFile);
+    if (checkFile.exists() && checkFile.isFile()) {
+        QFile::remove(pathToFile);
+    }
+
+
+    QJsonArray lstObjown;
+    QJsonObject objOwner;
+    for (auto obj = _ownersMap.cbegin(); obj != _ownersMap.end(); ++obj) {
+        objOwner["timePoint"] = obj.value().getTimePoint();
+        objOwner["name"] = obj.value().getOwnerName();
+
+        lstObjown.append(objOwner);
+    }
+
+    QJsonObject signJson;
+    signJson["ownersList"] = lstObjown;
+    signJson["license"] = _licenseTitle;
+    signJson["licenseText"] = _customMessage;
+
+    QByteArray jsonDoc;
+    jsonDoc = QJsonDocument(signJson).toJson();
+
+    QFile saveFile(pathToFile);
+    if (!saveFile.open(QIODevice::WriteOnly)) {
```

Comment:
Please add truncate mode for file. because the truncate mode force rewrite a opened file. see qt [documentation](https://doc.qt.io/qt-5/qiodevice.html#OpenModeFlag-enum) about open modes.
```suggestion
    if (!saveFile.open(QIODevice::WriteOnly | QIODevice::Truncate)) {
```